### PR TITLE
#460 Bring back implementation of Folded

### PR DESCRIPTION
**Pull request by bedward70**
_After the discussion_ "#460 Max and Min were renamed to HighestOf and LowestOf to support Comparable implementations #475".

**Step 1: Bring back implementation of `Folded`**
* The `Folded` class was deleted in the https://github.com/yegor256/cactoos/commit/25dc45d Yegor Bugayenko <yegor256@gmail.com> on 11/27/17 at 4:05 AM.
* I  brought the `Folded` class back.
* I  wrote the test case for `Folded`. Details: the `FoldedTest` class.

### CODE REVIEW

**Comment by 0crat** [comment]
@yegor256 please, pay attention to this pull request

**Comment by 0crat** [comment]
Job `gh:yegor256/cactoos#477` is in scope.

**Comment by 0crat** [comment]
Job `gh:yegor256/cactoos#477` assigned to @kkamkou, please go ahead ([policy](http://datum.zerocracy.com/pages/policy.html)).

**Comment by 0crat** [comment]
@kkamkou this job was assigned to you 5 days ago. It will be taken away from you soon, unless you close it, see [par.8](http://datum.zerocracy.com/pages/policy.html#8).

**Comment by yegor256** [comment]
@bedward70 can't you just provide `true` to `Reduced` if you don't really need the input? It seems that these two classes are duplicating each other: `Reduced` and `Folded`

**Comment by bedward70** [comment]
@yegor256 I double-checked and did not see the way to unite `Reduced` and `Folded`. 
I think we need these classes:
1. `Reduced` is a hoarder;
2. `Folded` is a selector of the best.

**Comment by 0crat** [comment]
@kkamkou this job was assigned to you 8 days ago. It will be taken away from you soon, unless you close it, see [par.8](http://datum.zerocracy.com/pages/policy.html#8).

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/datum):

```
java.lang.NumberFormatException: For input string: "97096402420866657c759a460ed0c282bcb3a555"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:583)
	at java.lang.Integer.parseInt(Integer.java:615)
	at com.zerocracy.radars.github.ReOnComment.react(ReOnComment.java:104)
	at com.zerocracy.radars.github.ReOnReason.react(ReOnReason.java:56)
	at com.zerocracy.radars.github.Reaction$Chain.react(Reaction.java:74)
	at com.zerocracy.radars.github.ReLogged.react(ReLogged.java:55)
	at com.zerocracy.radars.github.RbOnComment.react(RbOnComment.java:60)
	at com.zerocracy.radars.github.RbByActions.react(RbByActions.java:65)
	at com.zerocracy.radars.github.Rebound$Chain.react(Rebound.java:74)
	at com.zerocracy.radars.github.RbSafe.lambda$react$0(RbSafe.java:65)
	at org.cactoos.func.FuncWithFallback.apply(FuncWithFallback.java:299)
	at org.cactoos.func.IoCheckedFunc.lambda$apply$0(IoCheckedFunc.java:60)
	at org.cactoos.func....
```

[1.0-SNAPSHOT](https://github.com/zerocracy/farm/tree/1.0-SNAPSHOT) /cc @yegor256

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/datum):

```
java.lang.NumberFormatException: For input string: "97096402420866657c759a460ed0c282bcb3a555"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:583)
	at java.lang.Integer.parseInt(Integer.java:615)
	at com.zerocracy.radars.github.ReOnComment.react(ReOnComment.java:104)
	at com.zerocracy.radars.github.ReOnReason.react(ReOnReason.java:56)
	at com.zerocracy.radars.github.Reaction$Chain.react(Reaction.java:74)
	at com.zerocracy.radars.github.ReLogged.react(ReLogged.java:55)
	at com.zerocracy.radars.github.RbOnComment.react(RbOnComment.java:60)
	at com.zerocracy.radars.github.RbByActions.react(RbByActions.java:65)
	at com.zerocracy.radars.github.Rebound$Chain.react(Rebound.java:74)
	at com.zerocracy.radars.github.RbSafe.lambda$react$0(RbSafe.java:65)
	at org.cactoos.func.FuncWithFallback.apply(FuncWithFallback.java:299)
	at org.cactoos.func.IoCheckedFunc.lambda$apply$0(IoCheckedFunc.java:60)
	at org.cactoos.func....
```

[1.0-SNAPSHOT](https://github.com/zerocracy/farm/tree/1.0-SNAPSHOT) /cc @yegor256

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/datum):

```
java.lang.NumberFormatException: For input string: "97096402420866657c759a460ed0c282bcb3a555"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:583)
	at java.lang.Integer.parseInt(Integer.java:615)
	at com.zerocracy.radars.github.ReOnComment.react(ReOnComment.java:104)
	at com.zerocracy.radars.github.ReOnReason.react(ReOnReason.java:56)
	at com.zerocracy.radars.github.Reaction$Chain.react(Reaction.java:74)
	at com.zerocracy.radars.github.ReLogged.react(ReLogged.java:55)
	at com.zerocracy.radars.github.RbOnComment.react(RbOnComment.java:60)
	at com.zerocracy.radars.github.RbByActions.react(RbByActions.java:65)
	at com.zerocracy.radars.github.Rebound$Chain.react(Rebound.java:74)
	at com.zerocracy.radars.github.RbSafe.lambda$react$0(RbSafe.java:65)
	at org.cactoos.func.FuncWithFallback.apply(FuncWithFallback.java:299)
	at org.cactoos.func.IoCheckedFunc.lambda$apply$0(IoCheckedFunc.java:60)
	at org.cactoos.func....
```

[1.0-SNAPSHOT](https://github.com/zerocracy/farm/tree/1.0-SNAPSHOT) /cc @yegor256

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/datum):

```
java.lang.NumberFormatException: For input string: "97096402420866657c759a460ed0c282bcb3a555"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:583)
	at java.lang.Integer.parseInt(Integer.java:615)
	at com.zerocracy.radars.github.ReOnComment.react(ReOnComment.java:104)
	at com.zerocracy.radars.github.ReOnReason.react(ReOnReason.java:56)
	at com.zerocracy.radars.github.Reaction$Chain.react(Reaction.java:74)
	at com.zerocracy.radars.github.ReLogged.react(ReLogged.java:55)
	at com.zerocracy.radars.github.RbOnComment.react(RbOnComment.java:60)
	at com.zerocracy.radars.github.RbByActions.react(RbByActions.java:65)
	at com.zerocracy.radars.github.Rebound$Chain.react(Rebound.java:74)
	at com.zerocracy.radars.github.RbSafe.lambda$react$0(RbSafe.java:65)
	at org.cactoos.func.FuncWithFallback.apply(FuncWithFallback.java:299)
	at org.cactoos.func.IoCheckedFunc.lambda$apply$0(IoCheckedFunc.java:60)
	at org.cactoos.func....
```

[1.0-SNAPSHOT](https://github.com/zerocracy/farm/tree/1.0-SNAPSHOT) /cc @yegor256

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/datum):

```
java.lang.NumberFormatException: For input string: "97096402420866657c759a460ed0c282bcb3a555"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:583)
	at java.lang.Integer.parseInt(Integer.java:615)
	at com.zerocracy.radars.github.ReOnComment.react(ReOnComment.java:104)
	at com.zerocracy.radars.github.ReOnReason.react(ReOnReason.java:56)
	at com.zerocracy.radars.github.Reaction$Chain.react(Reaction.java:74)
	at com.zerocracy.radars.github.ReLogged.react(ReLogged.java:55)
	at com.zerocracy.radars.github.RbOnComment.react(RbOnComment.java:60)
	at com.zerocracy.radars.github.RbByActions.react(RbByActions.java:65)
	at com.zerocracy.radars.github.Rebound$Chain.react(Rebound.java:74)
	at com.zerocracy.radars.github.RbSafe.lambda$react$0(RbSafe.java:65)
	at org.cactoos.func.FuncWithFallback.apply(FuncWithFallback.java:299)
	at org.cactoos.func.IoCheckedFunc.lambda$apply$0(IoCheckedFunc.java:60)
	at org.cactoos.func....
```

[1.0-SNAPSHOT](https://github.com/zerocracy/farm/tree/1.0-SNAPSHOT) /cc @yegor256

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/datum):

```
java.lang.NumberFormatException: For input string: "97096402420866657c759a460ed0c282bcb3a555"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:583)
	at java.lang.Integer.parseInt(Integer.java:615)
	at com.zerocracy.radars.github.ReOnComment.react(ReOnComment.java:104)
	at com.zerocracy.radars.github.ReOnReason.react(ReOnReason.java:56)
	at com.zerocracy.radars.github.Reaction$Chain.react(Reaction.java:74)
	at com.zerocracy.radars.github.ReLogged.react(ReLogged.java:55)
	at com.zerocracy.radars.github.RbOnComment.react(RbOnComment.java:60)
	at com.zerocracy.radars.github.RbByActions.react(RbByActions.java:65)
	at com.zerocracy.radars.github.Rebound$Chain.react(Rebound.java:74)
	at com.zerocracy.radars.github.RbSafe.lambda$react$0(RbSafe.java:65)
	at org.cactoos.func.FuncWithFallback.apply(FuncWithFallback.java:299)
	at org.cactoos.func.IoCheckedFunc.lambda$apply$0(IoCheckedFunc.java:60)
	at org.cactoos.func....
```

[1.0-SNAPSHOT](https://github.com/zerocracy/farm/tree/1.0-SNAPSHOT) /cc @yegor256

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/datum):

```
java.lang.NumberFormatException: For input string: "97096402420866657c759a460ed0c282bcb3a555"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:583)
	at java.lang.Integer.parseInt(Integer.java:615)
	at com.zerocracy.radars.github.ReOnComment.react(ReOnComment.java:104)
	at com.zerocracy.radars.github.ReOnReason.react(ReOnReason.java:56)
	at com.zerocracy.radars.github.Reaction$Chain.react(Reaction.java:74)
	at com.zerocracy.radars.github.ReLogged.react(ReLogged.java:55)
	at com.zerocracy.radars.github.RbOnComment.react(RbOnComment.java:60)
	at com.zerocracy.radars.github.RbByActions.react(RbByActions.java:65)
	at com.zerocracy.radars.github.Rebound$Chain.react(Rebound.java:74)
	at com.zerocracy.radars.github.RbSafe.lambda$react$0(RbSafe.java:65)
	at org.cactoos.func.FuncWithFallback.apply(FuncWithFallback.java:299)
	at org.cactoos.func.IoCheckedFunc.lambda$apply$0(IoCheckedFunc.java:60)
	at org.cactoos.func....
```

[1.0-SNAPSHOT](https://github.com/zerocracy/farm/tree/1.0-SNAPSHOT) /cc @yegor256

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/datum):

```
java.lang.NumberFormatException: For input string: "97096402420866657c759a460ed0c282bcb3a555"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:583)
	at java.lang.Integer.parseInt(Integer.java:615)
	at com.zerocracy.radars.github.ReOnComment.react(ReOnComment.java:104)
	at com.zerocracy.radars.github.ReOnReason.react(ReOnReason.java:56)
	at com.zerocracy.radars.github.Reaction$Chain.react(Reaction.java:74)
	at com.zerocracy.radars.github.ReLogged.react(ReLogged.java:55)
	at com.zerocracy.radars.github.RbOnComment.react(RbOnComment.java:60)
	at com.zerocracy.radars.github.RbByActions.react(RbByActions.java:65)
	at com.zerocracy.radars.github.Rebound$Chain.react(Rebound.java:74)
	at com.zerocracy.radars.github.RbSafe.lambda$react$0(RbSafe.java:65)
	at org.cactoos.func.FuncWithFallback.apply(FuncWithFallback.java:299)
	at org.cactoos.func.IoCheckedFunc.lambda$apply$0(IoCheckedFunc.java:60)
	at org.cactoos.func....
```

[1.0-SNAPSHOT](https://github.com/zerocracy/farm/tree/1.0-SNAPSHOT) /cc @yegor256

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/datum):

```
java.lang.NumberFormatException: For input string: "97096402420866657c759a460ed0c282bcb3a555"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:583)
	at java.lang.Integer.parseInt(Integer.java:615)
	at com.zerocracy.radars.github.ReOnComment.react(ReOnComment.java:104)
	at com.zerocracy.radars.github.ReOnReason.react(ReOnReason.java:56)
	at com.zerocracy.radars.github.Reaction$Chain.react(Reaction.java:74)
	at com.zerocracy.radars.github.ReLogged.react(ReLogged.java:55)
	at com.zerocracy.radars.github.RbOnComment.react(RbOnComment.java:60)
	at com.zerocracy.radars.github.RbByActions.react(RbByActions.java:65)
	at com.zerocracy.radars.github.Rebound$Chain.react(Rebound.java:74)
	at com.zerocracy.radars.github.RbSafe.lambda$react$0(RbSafe.java:65)
	at org.cactoos.func.FuncWithFallback.apply(FuncWithFallback.java:299)
	at org.cactoos.func.IoCheckedFunc.lambda$apply$0(IoCheckedFunc.java:60)
	at org.cactoos.func....
```

[1.0-SNAPSHOT](https://github.com/zerocracy/farm/tree/1.0-SNAPSHOT) /cc @yegor256

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/datum):

```
java.lang.NumberFormatException: For input string: "97096402420866657c759a460ed0c282bcb3a555"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:583)
	at java.lang.Integer.parseInt(Integer.java:615)
	at com.zerocracy.radars.github.ReOnComment.react(ReOnComment.java:104)
	at com.zerocracy.radars.github.ReOnReason.react(ReOnReason.java:56)
	at com.zerocracy.radars.github.Reaction$Chain.react(Reaction.java:74)
	at com.zerocracy.radars.github.ReLogged.react(ReLogged.java:55)
	at com.zerocracy.radars.github.RbOnComment.react(RbOnComment.java:60)
	at com.zerocracy.radars.github.RbByActions.react(RbByActions.java:65)
	at com.zerocracy.radars.github.Rebound$Chain.react(Rebound.java:74)
	at com.zerocracy.radars.github.RbSafe.lambda$react$0(RbSafe.java:65)
	at org.cactoos.func.FuncWithFallback.apply(FuncWithFallback.java:299)
	at org.cactoos.func.IoCheckedFunc.lambda$apply$0(IoCheckedFunc.java:60)
	at org.cactoos.func....
```

[1.0-SNAPSHOT](https://github.com/zerocracy/farm/tree/1.0-SNAPSHOT) /cc @yegor256

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/datum):

```
java.lang.NumberFormatException: For input string: "97096402420866657c759a460ed0c282bcb3a555"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:583)
	at java.lang.Integer.parseInt(Integer.java:615)
	at com.zerocracy.radars.github.ReOnComment.react(ReOnComment.java:104)
	at com.zerocracy.radars.github.ReOnReason.react(ReOnReason.java:56)
	at com.zerocracy.radars.github.Reaction$Chain.react(Reaction.java:74)
	at com.zerocracy.radars.github.ReLogged.react(ReLogged.java:55)
	at com.zerocracy.radars.github.RbOnComment.react(RbOnComment.java:60)
	at com.zerocracy.radars.github.RbByActions.react(RbByActions.java:65)
	at com.zerocracy.radars.github.Rebound$Chain.react(Rebound.java:74)
	at com.zerocracy.radars.github.RbSafe.lambda$react$0(RbSafe.java:65)
	at org.cactoos.func.FuncWithFallback.apply(FuncWithFallback.java:299)
	at org.cactoos.func.IoCheckedFunc.lambda$apply$0(IoCheckedFunc.java:60)
	at org.cactoos.func....
```

[1.0-SNAPSHOT](https://github.com/zerocracy/farm/tree/1.0-SNAPSHOT) /cc @yegor256

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/datum):

```
java.lang.NumberFormatException: For input string: "97096402420866657c759a460ed0c282bcb3a555"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:583)
	at java.lang.Integer.parseInt(Integer.java:615)
	at com.zerocracy.radars.github.ReOnComment.react(ReOnComment.java:104)
	at com.zerocracy.radars.github.ReOnReason.react(ReOnReason.java:56)
	at com.zerocracy.radars.github.Reaction$Chain.react(Reaction.java:74)
	at com.zerocracy.radars.github.ReLogged.react(ReLogged.java:55)
	at com.zerocracy.radars.github.RbOnComment.react(RbOnComment.java:60)
	at com.zerocracy.radars.github.RbByActions.react(RbByActions.java:65)
	at com.zerocracy.radars.github.Rebound$Chain.react(Rebound.java:74)
	at com.zerocracy.radars.github.RbSafe.lambda$react$0(RbSafe.java:65)
	at org.cactoos.func.FuncWithFallback.apply(FuncWithFallback.java:299)
	at org.cactoos.func.IoCheckedFunc.lambda$apply$0(IoCheckedFunc.java:60)
	at org.cactoos.func....
```

[1.0-SNAPSHOT](https://github.com/zerocracy/farm/tree/1.0-SNAPSHOT) /cc @yegor256

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/datum):

```
java.lang.NumberFormatException: For input string: "97096402420866657c759a460ed0c282bcb3a555"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:583)
	at java.lang.Integer.parseInt(Integer.java:615)
	at com.zerocracy.radars.github.ReOnComment.react(ReOnComment.java:104)
	at com.zerocracy.radars.github.ReOnReason.react(ReOnReason.java:56)
	at com.zerocracy.radars.github.Reaction$Chain.react(Reaction.java:74)
	at com.zerocracy.radars.github.ReLogged.react(ReLogged.java:55)
	at com.zerocracy.radars.github.RbOnComment.react(RbOnComment.java:60)
	at com.zerocracy.radars.github.RbByActions.react(RbByActions.java:65)
	at com.zerocracy.radars.github.Rebound$Chain.react(Rebound.java:74)
	at com.zerocracy.radars.github.RbSafe.lambda$react$0(RbSafe.java:65)
	at org.cactoos.func.FuncWithFallback.apply(FuncWithFallback.java:299)
	at org.cactoos.func.IoCheckedFunc.lambda$apply$0(IoCheckedFunc.java:60)
	at org.cactoos.func....
```

[1.0-SNAPSHOT](https://github.com/zerocracy/farm/tree/1.0-SNAPSHOT) /cc @yegor256

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
@allanws[/z](http://www.0crat.com/u/allanws) resigned from [#477](https://github.com/yegor256/cactoos/issues/477), please stop working

**Comment by 0crat** [comment]
The job [#477](https://github.com/yegor256/cactoos/issues/477) assigned to @llorllale[/z](http://www.0crat.com/u/llorllale). The budget is 30 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§8](http://datum.zerocracy.com/pages/policy.html#8) and [§9](http://datum.zerocracy.com/pages/policy.html#9). If the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html).

File: src/main/java/org/cactoos/scalar/Folded.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+     * Folding function.
+     */
+    private final BiFunc<T, T, T> fold;
+
+    /**
+     * Ctor.
+     * @param fold Folding function
+     * @param items The items
+     */
+    public Folded(final BiFunc<T, T, T> fold, final Iterable<Scalar<T>> items) {
```
Comment (llorllale) [inline src/main/java/org/cactoos/scalar/Folded.java]: Parameter names must be different from attribute names.

File: src/main/java/org/cactoos/scalar/Folded.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+    @Override
+    public T value() throws Exception {
+        final Iterator<Scalar<T>> iter = this.items.iterator();
+        if (!iter.hasNext()) {
+            throw new NoSuchElementException(
+                "Can't find first element in an empty iterable"
+            );
+        }
+        T acc = iter.next().value();
```
Comment (llorllale) [inline src/main/java/org/cactoos/scalar/Folded.java]: Does this pass qulice? In any case, all local variables must be declared `final`

File: src/main/java/org/cactoos/scalar/Folded.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        this.items = items;
+        this.fold = fold;
+    }
+
+    @Override
+    public T value() throws Exception {
+        final Iterator<Scalar<T>> iter = this.items.iterator();
+        if (!iter.hasNext()) {
+            throw new NoSuchElementException(
+                "Can't find first element in an empty iterable"
```
Comment (llorllale) [inline src/main/java/org/cactoos/scalar/Folded.java]: Why not?

File: src/main/java/org/cactoos/scalar/HighestOf.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+ * Find the greater among items.
+ *
+ * <p>This class implements {@link Scalar}, which throws a checked
+ * {@link Exception}. This may not be convenient in many cases. To make
+ * it more convenient and get rid of the checked exception you can
+ * use {@link UncheckedScalar} or {@link IoCheckedScalar} decorators.</p>
+ *
+ * <p>There is no thread-safety guarantee.
+ *
+ * @author Fabricio Cabral (fabriciofx@gmail.com)
```
Comment (llorllale) [inline src/main/java/org/cactoos/scalar/HighestOf.java]: The @author now would be you, @bedward70

File: src/main/java/org/cactoos/scalar/LowestOf.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+ * Find the smaller among items.
+ *
+ * <p>There is no thread-safety guarantee.
+ *
+ * <p>This class implements {@link Scalar}, which throws a checked
+ * {@link Exception}. This may not be convenient in many cases. To make
+ * it more convenient and get rid of the checked exception you can
+ * use {@link UncheckedScalar} or {@link IoCheckedScalar} decorators.</p>
+ *
+ * @author Fabricio Cabral (fabriciofx@gmail.com)
```
Comment (llorllale) [inline src/main/java/org/cactoos/scalar/LowestOf.java]: You should be the @author

File: src/main/java/org/cactoos/scalar/Folded.java (hunk 3 above)
Comment (bedward70) [reply src/main/java/org/cactoos/scalar/Folded.java]: * The `Folded` class was deleted in the https://github.com/yegor256/cactoos/commit/25dc45d Yegor Bugayenko <yegor256@gmail.com> on 11/27/17 at 4:05 AM.
So, I restored the `Folded` class without changes.
* "I made MaxFunc and MinFunc to inline". See 
https://github.com/yegor256/cactoos/pull/477/commits/eb5ed804056d8ee7e7549494ad9f7676c2cad42e

File: src/main/java/org/cactoos/scalar/Folded.java (hunk 2 above)
Comment (bedward70) [reply src/main/java/org/cactoos/scalar/Folded.java]: Yes, qulice passed it. See 3 lines down: `acc = this.function.apply(acc, next);`

File: src/main/java/org/cactoos/scalar/HighestOf.java (hunk 4 above)
Comment (bedward70) [reply src/main/java/org/cactoos/scalar/HighestOf.java]: @llorllale
The `Max` class was deleted in the https://github.com/yegor256/cactoos/commit/25dc45d Yegor Bugayenko <yegor256@gmail.com> on 11/27/17 at 4:05 AM.
I restored the `Max` and rename it to `HighestOf`.
Therefore the class was created by `@author Fabricio Cabral (fabriciofx@gmail.com)`.

File: src/main/java/org/cactoos/scalar/LowestOf.java (hunk 5 above)
Comment (bedward70) [reply src/main/java/org/cactoos/scalar/LowestOf.java]: @llorllale
The `Min` class was deleted in the https://github.com/yegor256/cactoos/commit/25dc45d Yegor Bugayenko <yegor256@gmail.com> on 11/27/17 at 4:05 AM.
I restored the `Min` and rename it to `LowestOf`.
Therefore the class was created by `@author Fabricio Cabral (fabriciofx@gmail.com)`.

**Comment by bedward70** [comment]
@llorllale The findings were commented or fixed by @bedward70.

**Comment by llorllale** [comment]
@bedward70 thank you for the explanations.

This PR is still too big - definitely does not fit in our "15 mins review window". It needs to be broken into 3 PRs:
1. This PR could be just "Bring back implementation of `Folded`". Make references to the comment thread in #460 to add context, reference the specific commit where it was deleted like you've done here, and also include tests (which you haven't done here). I expect 2 files to be delivered.
2. "Bring back `Max` implementation but rename to `HighestOf`". Make the necessary references to provide context like the previous point, and also add your own @author tag next to Fabricio's (since it's technically a new class). Deliverables: 2 files (class + test)
3. "Bring back `Min` implementation but rename to `LowestOf`". Same as above. Deliverables: 2 files.

**Comment by llorllale** [comment]
@bedward70 are you working on this PR?

**Comment by bedward70** [comment]
@llorllale I have done. I split the PR to 3 PRs:
1. #460 Bring back implementation of Folded: #477. 
2. #460 Bring back Max implementation but rename to HighestOf: #575;
3. #460 Bring back Min implementation but rename to LowestOf: #576.

File: src/test/java/org/cactoos/scalar/FoldedTest.java [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * Test case for {@link Folded}.
+ *
+ * @author Eduard Balovnev (bedward70@mail.ru)
+ * @version $Id$
+ * @since 0.10
```
Comment (llorllale) [inline src/test/java/org/cactoos/scalar/FoldedTest.java]: please change to `1.0`

File: src/test/java/org/cactoos/scalar/FoldedTest.java [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+ * @checkstyle JavadocMethodCheck (500 lines)
+ */
+public final class FoldedTest {
+
+    @Test(expected = NoSuchElementException.class)
+    public void emptyListTest() throws Exception {
```
Comment (llorllale) [inline src/test/java/org/cactoos/scalar/FoldedTest.java]: Take a look at the method names of other tests in cactoos to get an idea of how test methods should be named. For example, this one could be called "failsForEmptyIterable"

File: src/test/java/org/cactoos/scalar/FoldedTest.java [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+    }
+
+    @Test
+    public void singleTest() throws Exception {
+        final Integer single = 10;
+        MatcherAssert.assertThat(
+            "Can't find the single",
+            new Folded<>(
+                (first, last) -> first,
+                new IterableOf<Scalar<Integer>>((Scalar<Integer>) () -> single)
```
Comment (llorllale) [inline src/test/java/org/cactoos/scalar/FoldedTest.java]: There's a lot of "generics" noise in this line (and several others further down). Can the diamond operator be used?

File: src/test/java/org/cactoos/scalar/FoldedTest.java [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+            )
+                .value(),
```
Comment (llorllale) [inline src/test/java/org/cactoos/scalar/FoldedTest.java]: can you attach this `.value()` to the closing parenthesis?

**Comment by llorllale** [comment]
Thank you @bedward70, looks better. Check the comments I just dropped now for a few details regarding the test class.

**Comment by llorllale** [comment]
@bedward70 are you working on this?

**Comment by bedward70** [comment]
@llorllale I can't check after `merge master`. See details:


java.lang.AssertionError: Can't calculate the file's MD5 checksum
Expected: Text with "842a5e7012d76e1df96c3d92e5c661df"
     but: was <org.cactoos.text.HexOf@2173f6d9>

	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.cactoos.io.Md5DigestOfTest.checksumFromFile(Md5DigestOfTest.java:74)

java.lang.AssertionError: Can't calculate the file's SHA-1 checksum
Expected: Text with "9d47e35afdcbf845aa9f05f15b4d936b97e55f0e"
     but: was <org.cactoos.text.HexOf@4abdb505>

	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.cactoos.io.Sha1DigestOfTest.checksumFromFile(Sha1DigestOfTest.java:74)
java.lang.AssertionError: Can't calculate the file's SHA-256 checksum
Expected: Text with "a56c3be45f9be8dda0653e33ae7ef3abf2939f926eda801f329e0830b6e7cc22"
     but: was <org.cactoos.text.HexOf@29b5cd00>

	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.cactoos.io.Sha256DigestOfTest.checksumFromFile(Sha256DigestOfTest.java:76)

**Comment by llorllale** [comment]
@bedward70 you're right - `master` is broken

**Comment by llorllale** [comment]
@bedward70 I've opened #580 for this. In the meanwhile, can you work around the error? Perhaps temporarily `@Ignore` the tests? Just make sure not to commit it!

**Comment by bedward70** [comment]
@llorllale I have fixed the findings. Thank you for your patience. :)

**Comment by llorllale** [comment]
@yegor256 please merge

**Comment by 0crat** [comment]
The user @llorllale[/z](https://www.0crat.com/u/llorllale) resigned from [#477](https://github.com/yegor256/cactoos/issues/477), please stop working

**Comment by 0crat** [comment]
The job [#477](https://github.com/yegor256/cactoos/issues/477) assigned to @driver733[/z](https://www.0crat.com/u/driver733). The budget is 30 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§8](http://datum.zerocracy.com/pages/policy.html#8) and [§9](http://datum.zerocracy.com/pages/policy.html#9). If the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html).

**Comment by driver733** [comment]
@yegor256 I believe this PR is ready to be merged as per the discussion.
https://github.com/yegor256/cactoos/pull/477#issuecomment-360859649

**Comment by driver733** [comment]
@0crat waiting

**Comment by 0crat** [comment]
> @0crat waiting ([here](https://github.com/yegor256/cactoos/issues/477#issuecomment-361100472))

@driver733 The impediment for [#477](https://github.com/yegor256/cactoos/issues/477) was registered successfully by @driver733[/z](https://www.0crat.com/u/driver733)

**Comment by yegor256** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/yegor256/cactoos/issues/477#issuecomment-361229802))

@yegor256 This is what I know about this job, as in [§32](http://datum.zerocracy.com/pages/policy.html#32):

  * The job [#477](https://github.com/yegor256/cactoos/issues/477) is [in scope](http://datum.zerocracy.com/pages/policy.html#14) for 2mon
  * The role is `DEV`
  * The job is assigned to @driver733[/z](https://www.0crat.com/u/driver733) for [2days](http://datum.zerocracy.com/pages/policy.html#8)
  * There is a monetary reward attached
  * The job has an [impediment](http://datum.zerocracy.com/pages/policy.html#9)
  * The [budget](http://datum.zerocracy.com/pages/policy.html#4) is 30 minutes
  * These users are banned and won't be assigned:
    * @bedward70[/z](https://www.0crat.com/u/bedward70): This user reported the ticket
    * @allanws[/z](https://www.0crat.com/u/allanws): User was resigned from the ticket
    * @llorllale[/z](https://www.0crat.com/u/llorllale): User was resigned from the ticket

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
Order was successfully finished: +30 points just awarded to @driver733[/z](https://www.0crat.com/u/driver733), total is [+30](https://www.0crat.com/u/driver733)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://datum.zerocracy.com/pages/policy.html#28): +15 points just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256), total is [+8670](https://www.0crat.com/u/yegor256)

**Comment by 0crat** [comment]
The job [#477](https://github.com/yegor256/cactoos/issues/477) is now out of scope

**Pull request merged**